File: m2-h2/src/main/java/ru/yegorpilipenko/otus/spring/m2h2/service/GenreService.java
package ru.yegorpilipenko.otus.spring.m2h2.service;

import ru.yegorpilipenko.otus.spring.m2h2.document.Genre;

import javax.annotation.Nonnull;
import java.util.Collection;

public interface GenreService {

    @Nonnull
    Collection<Genre> getAll();

}

File: m1-h5/src/main/kotlin/ru/yegorpilipenko/otus/spring/m1h5/service/GenreService.kt
package ru.yegorpilipenko.otus.spring.m1h5.service

import ru.yegorpilipenko.otus.spring.m1h5.model.Genre

interface GenreService {

    fun findAll(): Collection<Genre>

    fun findById(id: Long): Genre?

    fun save(genre: Genre): Genre

    fun removeById(id: Long): Boolean
}